plugin-cgroupv2/src/k8s/token.rs

```diff
@@ -55,70 +55,65 @@ impl Token {
+                    .context("ERROR : kubectl command execution")?;
 
-                if !output.status.success() {
-                    return Err(anyhow!(
-                        "kubectl raised an error: {}",
-                        String::from_utf8_lossy(&output.stderr)
-                    ));
-                }
-
-                String::from_utf8_lossy(&output.stdout).trim().to_string()
+                String::from_utf8(output.stdout)
+                    .context("ERROR : UTF-8 output conversion")?
+                    .trim()
+                    .to_string()
             }
+
             TokenRetrieval::File => {
                 let mut file = File::open("/var/run/secrets/kubernetes.io/serviceaccount/token")
-                    .map_err(|e| anyhow!("Failed to open service account token file: {}", e))?;
+                    .context("ERROR : opening token file")?;
                 let mut token = String::new();
-                file.read_to_string(&mut token)
-                    .map_err(|e| anyhow!("Failed to read service account token file: {}", e))?;
+                file.read_to_string(&mut token).expect("ERROR : Reading token file");
                 token
             }
         };
 
-        let mut components = token.split('.');
-        let payload = components
+        let mut components: std::str::Split<'_, char> = token.split('.');
+        let payload: &str = components
             .next()
-            .ok_or_else(|| anyhow!("Could not parse the token, payload is missing"))?;
+            .context("ERROR : Missing payload, token can be parsed")?;
 
-        let decoded_payload = BASE64_STANDARD_NO_PAD
+        let decoded_payload: Vec<u8> = BASE64_STANDARD_NO_PAD
             .decode(payload.trim())
-            .map_err(|e| anyhow!("Failed to decode payload: {}", e))?;
+            .context("ERROR : payload decoding")?;
 
         let payload_json: serde_json::Value =
-            serde_json::from_slice(&decoded_payload).map_err(|e| anyhow!("Failed to parse payload as JSON: {}", e))?;
-
-        let exp = match payload_json.get("exp") {
-            Some(exp) => exp
-                .as_u64()
-                .ok_or_else(|| anyhow!("Expiration time 'exp' is not a valid u64"))?,
-            None => {
-                let mut new_data = self.data.write().await;
-                new_data.expiration_time = None;
-                return Err(anyhow!("Could not extract the 'exp' field from the token"));
-            }
-        };
+            serde_json::from_slice(&decoded_payload).context("ERROR : JSON payload parsing")?;
+
+        let exp: u64 = payload_json
+            .get("exp")
+            .and_then(|exp| exp.as_u64())
+            .context("ERROR : Missing or invalid 'exp' field in token")?;
 
         {
-            let mut new_data = self.data.write().await;
+            let mut new_data: tokio::sync::RwLockWriteGuard<'_, TokenData> = self.data.write().await;
```

Comment:
Please don't add explicit types in this situation, it's only creating more "diffs" to review.